### ISSUE TITLE
File is not closed on XML object destruction

### ISSUE THREAD

**Issue by xanview** [sophsec/ruby-nmap#45]
If you do:

```
results = Nmap::XML.new('/tmp/tmp.xml')
results = nil
```

The file is kept open in lsof. If you are analysing hundreds of results, it keeps hundreds of files open :( 

```
$ lsof -n | grep -i tmp.xml
ruby2.3    18083               deployer   14r      REG                8,1        0    1131267 /tmp/tmp.xml
logger.rb  18083  18097        deployer   14r      REG                8,1        0    1131267 /tmp/tmp.xml
spawner.r  18083  18102        deployer   14r      REG                8,1        0    1131267 /tmp/tmp.xml
spawner.r  18083  18103        deployer   14r      REG                8,1        0    1131267 /tmp/tmp.xml
spawner.r  18083  18104        deployer   14r      REG                8,1        0    1131267 /tmp/tmp.xml
spawner.r  18083  18106        deployer   14r      REG                8,1        0    1131267 /tmp/tmp.xml
ruby-time  18083  18108        deployer   14r      REG                8,1        0    1131267 /tmp/tmp.xml
```

Any solution?

**Comment by postmodern** [comment]
Off the top of my head `open(path) { |file| @doc = Nokogiri::XML(file) }`. I want to avoid loading the full file contents into memory, as some nmap XML files can get quite large. Nokogiri appears to support streamed parsing of IO objects under the hood. https://github.com/sparklemotion/nokogiri/blob/7f8b7b2bf55829e02cd11c2eb25814c3ed458676/ext/nokogiri/xml_document.c#L219-L267

**Comment by postmodern** [comment]
Fixed in 24a99ebf7e3a48fa37251b6ee5babce970e10064. Releasing 0.9.2 shortly.
